src/writer/core.py

```
@@ -2305,28 +2456,28 @@ def _deserialize_bigint_format(payload: Optional[Union[dict, list]]):
     if (
-        expected_type is not None
-        and inspect.isclass(expected_type)
-        and issubclass(expected_type, dict)
+        type_to_check is not None
+        and inspect.isclass(type_to_check)
+        and issubclass(type_to_check, expected_type)
     ):
         return True
 
-    if typing.get_origin(expected_type) == dict:
+    if typing.get_origin(type_to_check) == dict:
```

Ruff (0.12.2)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)